geomesa-spark-jts/src/main/scala/org/apache/spark/sql/jts/SQLRules.scala

```diff
@@ -0,0 +1,63 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.apache.spark.sql.jts
+
+import com.vividsolutions.jts.geom.Geometry
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.expressions.{GenericInternalRow, LeafExpression, Literal, ScalaUDF}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.jts.JTSTypes._
+import scala.util.Try
+
+object SQLRules {
+
+  // new AST expressions
+  case class GeometryLiteral(repr: InternalRow, geom: Geometry) extends LeafExpression  with CodegenFallback {
+
+    override def foldable: Boolean = true
+
+    override def nullable: Boolean = true
+
+    override def eval(input: InternalRow): Any = repr
+
+    override def dataType: DataType = GeometryTypeInstance
+  }
+
+
+  object ScalaUDFRule extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = {
+      plan.transform {
+        case q: LogicalPlan => q.transformExpressionsDown {
+          case s@ScalaUDF(_, _, _, _, _) =>
```

Comment:
can we change this to be `s: ScalaUDF`? we have an incompatibility in spark 2.0 vs 2.2 with an unapply pattern like this (as the case class changed its number of variables), and in this case we're not actually using the unapplied values so it doesn't seem necessary.